source/particle/particle_handler.cc

```diff
@@ -0,0 +1,194 @@
+/*
+  Copyright (C) 2015 - 2017 by the authors of the ASPECT code.
+
+  This file is part of ASPECT.
+
+  ASPECT is free software; you can redistribute it and/or modify
+  it under the terms of the GNU General Public License as published by
+  the Free Software Foundation; either version 2, or (at your option)
+  any later version.
+
+  ASPECT is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with ASPECT; see the file LICENSE.  If not see
+  <http://www.gnu.org/licenses/>.
+*/
+
+#include <aspect/particle/particle_handler.h>
+
+namespace aspect
+{
+  namespace Particle
+  {
+    template <int dim,int spacedim>
+    ParticleHandler<dim,spacedim>::ParticleHandler()
+      :
+      triangulation(),
+      mpi_communicator(),
+      global_number_of_particles(0),
+      global_max_particles_per_cell(0),
+      next_free_particle_index(0)
+    {}
+
+    template <int dim,int spacedim>
+    ParticleHandler<dim,spacedim>::ParticleHandler(const parallel::distributed::Triangulation<dim,spacedim> &triangulation,
+                                                   MPI_Comm mpi_communicator)
+      :
+      triangulation(&triangulation, typeid(*this).name()),
+      mpi_communicator(mpi_communicator),
+      global_number_of_particles(0),
+      global_max_particles_per_cell(0),
+      next_free_particle_index(0)
+    {}
+
+    template <int dim,int spacedim>
+    ParticleHandler<dim,spacedim>::~ParticleHandler()
+    {}
+
+    template <int dim,int spacedim>
+    void
+    ParticleHandler<dim,spacedim>::clear()
+    {
+      particles.clear();
+      global_number_of_particles = 0;
+      next_free_particle_index = 0;
+      global_max_particles_per_cell = 0;
+    }
+
+    template <int dim,int spacedim>
+    typename ParticleHandler<dim,spacedim>::particle_iterator
+    ParticleHandler<dim,spacedim>::begin() const
+    {
+      return particle_iterator(particles,(const_cast<ParticleHandler<dim,spacedim> *> (this))->particles.begin());
+    }
+
+    template <int dim,int spacedim>
```

Comment:
since you're already editing around these files, please make it 3 empty lines between functions -- that's the deal.II style